### PR TITLE
Self taught evaluators project release

### PR DESCRIPTION
**Pull request by uralik**
We release

* evaluation and inference instructions
* data generation pipeline scripts
* dataset loading example
* training configs and fairseq2 commands for training

### CODE REVIEW

**Comment by janedwivedi** [comment]
Should requirements.txt be in the highest dir level? Or maybe change "Please refer to [src/requirements.txt](./src/requirements.txt)"?

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -7,7 +7,7 @@ repos:
     - id: no-commit-to-branch
       args: ['--branch', 'main']
     - id: check-added-large-files
-      args: ['--maxkb=2000']
+      args: ['--maxkb=20000']
```
Comment (Golovneva) [inline .pre-commit-config.yaml:10]: Why is it necessary? Are you committing any especially large files?

Comment (uralik) [reply .pre-commit-config.yaml:10]: yes, rewardbench inputs are >2MB , so I changed to 20

File: projects/self_taught_evaluator/README.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
+# Self-Taught Evaluators
+
+<p align="center"><img width="90%" src="figures/self_taught_dpo.png" /></p>
+
+Instructions and materials presented here correspond to the [Self-taught evaluators](https://arxiv.org/abs/2408.02666) research project.
```
Comment (Golovneva) [inline projects/self_taught_evaluator/README.md:5]: research project -> research paper?
this links to paper

File: projects/self_taught_evaluator/README.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+## Synthetic Preference Data
+
+The pre-processed training data for preference fine-tuning can be downloaded here: https://huggingface.co/datasets/facebook/Self-taught-evaluator-DPO-data
+
+Below you can find instructions on how to replicate our data generation process.
+
+### Generate worse response
+1. Given pairs of (instruction, baseline response), prepare prompts using the template specified in `data/prompts/worse_response.prompt`.
+2. Run generation on the prompts from step 1, to generate a "worse response" to the instruction.
```
Comment (Golovneva) [inline projects/self_taught_evaluator/README.md:47]:
```suggestion
2. Run generation on the prompts from step 1, to generate a "worse response" to the instruction.

```

File: projects/self_taught_evaluator/README.md (hunk 2 above)
Comment (uralik) [reply projects/self_taught_evaluator/README.md:5]:
```suggestion
Instructions and materials presented here correspond to the [Self-taught evaluators](https://arxiv.org/abs/2408.02666) research paper.
```

**Review by Golovneva — APPROVED** [review]
LGTM!

**Pull request merged**